frontend/document/credocument.lua

```diff
@@ -1667,8 +1663,8 @@ function CreDocument:setupCallCache()
             elseif cache_by_tag then
                 is_cached = true
                 self[name] = function(...)
-                    local starttime = TimeVal:now()
-                    local cache_key = name .. asString(select(2, ...))
+                    local starttime = now()
+                    local cache_key = name .. buffer.encode({select(2, ...)})
```

Comment:
OK, replaced with this new LuaJIT string.buffer thingy.
Guess you checked with the call cache stats that they indeed give repeatable same key string and everything still works :)

Reply:
Pending some more checking later tonight, but it's not worse, no.

The only question mark left is key ordering in hashes, but given that the old code was already using `pairs`, at best it's a "broken" as before ;).

At first glance, I get as stable results as before, though (we might be safe-ish, given that the context of functions that take a Geom is very restricted, so we might more-or-less hit "stable" ordering in relevant contexts).

Reply:
> given that the old code was already using pairs, at best it's a "broken" as before ;).

Oups :)

Reply:
On that note: does it even make sense to cache those `getLinkFromPosition` calls? it'd take some truly lucky tap to get a cache hit, wouldn't it?

Reply:
Yeah, tested with the `configurable` hash used in `koptinterface`, and it's indeed as unstable as `pairs` (i.e., it *is* `pairs`, essentially).

That said, given the much more restricted context (same process, same page), I *think* we're mostly okay.